src/main/java/com/synopsys/integration/alert/component/settings/actions/SettingsGlobalApiAction.java

```diff
@@ -75,96 +76,111 @@ public FieldModel afterGetAction(final FieldModel fieldModel) {
     }
 
     @Override
-    public FieldModel beforeSaveAction(final FieldModel fieldModel) {
+    public FieldModel beforeSaveAction(FieldModel fieldModel) {
         return handleNewAndUpdatedConfig(fieldModel);
     }
 
     @Override
-    public FieldModel beforeUpdateAction(final FieldModel fieldModel) {
+    public FieldModel beforeUpdateAction(FieldModel fieldModel) {
         return handleNewAndUpdatedConfig(fieldModel);
     }
 
-    private FieldModel createFieldModelCopy(final FieldModel fieldModel) {
-        final HashMap<String, FieldValueModel> fields = new HashMap<>();
+    private FieldModel createFieldModelCopy(FieldModel fieldModel) {
+        HashMap<String, FieldValueModel> fields = new HashMap<>();
         fields.putAll(fieldModel.getKeyToValues());
 
-        final FieldModel modelToSave = new FieldModel(fieldModel.getDescriptorName(), fieldModel.getContext(), fields);
+        FieldModel modelToSave = new FieldModel(fieldModel.getDescriptorName(), fieldModel.getContext(), fields);
         modelToSave.setId(fieldModel.getId());
         return modelToSave;
     }
 
-    private FieldModel handleNewAndUpdatedConfig(final FieldModel fieldModel) {
+    @Override
+    public FieldModel afterSaveAction(FieldModel fieldModel) throws AlertException {
+        addSAMLMetadata(fieldModel);
```

Comment:
Do we need to call `addSAMLMetadata(fieldModel);` in both the afterSaveAction and in the afterUpdateAction?

Reply:
yes the save is for a POST and update is for the PUT rest call.